src/main/java/frc/team2412/robot/commands/drivebase/DriveCommand.java

```diff
@@ -0,0 +1,97 @@
+package frc.team2412.robot.commands.drivebase;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.networktables.GenericEntry;
+import edu.wpi.first.wpilibj.shuffleboard.BuiltInWidgets;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.team2412.robot.subsystems.DrivebaseSubsystem;
+import java.util.Map;
+import java.util.function.DoubleSupplier;
+
+public class DriveCommand extends Command {
+	private static final double TURBO_ROTATION_DEFAULT = 1.5;
+
+	private final DrivebaseSubsystem drivebaseSubsystem;
+	private final DoubleSupplier forward;
+	private final DoubleSupplier strafe;
+	private final DoubleSupplier rotation;
+	private final DoubleSupplier turboRotation;
+	// temporary because code is kinda messy
+	Rotation2d MAX_ROTATIONS_PER_SEC = Rotation2d.fromRotations(0.8574);
+
+	// shuffleboard
+	private static GenericEntry driveSpeedEntry =
+			Shuffleboard.getTab("Drivebase")
+					.addPersistent("Drive Speed 1", 1)
+					.withSize(2, 1)
+					.withWidget(BuiltInWidgets.kNumberSlider)
+					.withProperties(Map.of("Min", 0, "Max", 1))
+					.getEntry();
+	private static GenericEntry rotationSpeedEntry =
+			Shuffleboard.getTab("Drivebase")
+					.addPersistent("Rotation Speed 1", 1)
+					.withSize(2, 1)
+					.withWidget(BuiltInWidgets.kNumberSlider)
+					.withProperties(Map.of("Min", 0, "Max", 1))
+					.getEntry();
+	private static GenericEntry fieldOrientedEntry =
+			Shuffleboard.getTab("Drivebase")
+					.addPersistent("Field Oriented 1", true)
+					.withWidget(BuiltInWidgets.kToggleSwitch)
+					.getEntry();
+	private static GenericEntry turboRotationEntry =
+			Shuffleboard.getTab("Drivebase")
+					.addPersistent("Turbo Rotation Modifier 1", TURBO_ROTATION_DEFAULT)
+					.withWidget(BuiltInWidgets.kNumberSlider)
+					.withProperties(Map.of("Min", 0.5, "Max", 2.5))
+					.getEntry();
+
+	public DriveCommand(
+			DrivebaseSubsystem drivebaseSubsystem,
+			DoubleSupplier forward,
+			DoubleSupplier strafe,
+			DoubleSupplier rotation,
+			DoubleSupplier turboRotation) {
+		this.drivebaseSubsystem = drivebaseSubsystem;
+		this.forward = forward;
+		this.strafe = strafe;
+		this.rotation = rotation;
+		// this variable give the right trigger input
```

Comment:
Delete, as this will quickly get out of sync with the control mapping

Reply:
Outdated because of new push and logic restructuring